components/mpas-seaice/src/shared/mpas_seaice_icepack.F

```diff
@@ -367,6 +382,258 @@ subroutine init_column_itd(domain)
 
   end subroutine init_column_itd
 
+
+
+!|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
+!
+!  init_column_fsd
+!
+!> \brief
+!> \author Erin E. Thomas, LANL
+!> \date Feb 2023
+!> \details
+!>
+!
+!-----------------------------------------------------------------------
+
+  subroutine init_column_fsd(domain)
+!    
+    use icepack_intfc, only: &
+            icepack_init_fsd_bounds, &
+            icepack_init_fsd, &
+            icepack_cleanup_fsd, &
+            icepack_char_len_long, &
+            c0, &
+            c1
+    use seaice_constants, only: &
+         seaicePuny
+
+    type(domain_type), intent(inout) :: domain
+
+    type(block_type), pointer :: block
+
+    type(MPAS_pool_type), pointer :: &
+         mesh, &
+         tracers, &
+         floe_size_distribution
+
+    logical, pointer :: config_use_column_waves
+    logical, pointer :: config_use_floe_size_distribution
+
+    ! dimensions
+    integer, pointer ::   &
+         nCellsSolve,     &
+         nCategories, &
+         nFloeCategories
+
+    !variables
+    real(kind=RKIND), dimension(:), pointer :: &
+         FloeSizeBinLower,   & !ndim = nFloeCat
+         FloeSizeBinCenter,  &
+         FloeSizeBinWidth, &
+         FloeDist
+
+    real(kind=RKIND), dimension(:,:), pointer :: &
+         DFloeSizeNewIce, & ! ndim = nFloeCat,nCells
+         DFloeSizeLateralGrowth, &
+         DFloeSizeLateralMelt, &
+         DFloeSizeWaves, &
+         DFloeSizeWeld, &
+         FloeDistCat
+
+    real(kind=RKIND), dimension(:,:,:), pointer :: &
+         iceAreaCategory, &
+         floeSizeDist !ndim = nFloeCat,nCat,NCells
+
+    !local variables
+    integer :: i,iCell, iCategory, iFloeCategory
+    character (len=35) :: c_fsd_range 
+    character (len=icepack_char_len_long) :: ice_init_type
+
+    call MPAS_pool_get_config(domain % configs, "config_use_column_waves", config_use_column_waves)
+    call MPAS_pool_get_config(domain % configs, "config_use_floe_size_distribution",config_use_floe_size_distribution)
+   
+    ice_init_type='default' ! NOTE: this should probably be a namelist setting? how to define for restarts?
```

Comment:
The namelist variable `ice_ic` in Icepack (and CICE) is overloaded in the sense that it can be used to specify whether a run will use a restart file or start from scratch, and if it's starting from scratch it specifies how the ice is configured, and if it's starting from a restart then it gives the name of the restart file (and yes I think this drives some of our users/developers a little batty).  MPAS-SI divides all of that functionality up into separate flags.  E.g. there's `config_initial_condition_type` in mpas_seaice_initialize.F, and there's `config_do_restart` and `runtype` in ice_comp_mct.F, so I think you'll just need to grab the ones you need and not worry about defining `ice_ic` separately.